modules/react/button/lib/Button.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import {BaseButton, ButtonContainerProps} from './BaseButton';
+import {createComponent} from '@workday/canvas-kit-react/common';
+
+/**
+ * Extends all the style properties from Box to our buttons as well as props from ButtonContainerProps.
+ * We omit `ref` since all of our buttons use `createComponent` and already give access to `ref`.
+ * Use this type to extend and customize any one off buttons that you want full control over styling.
+ */
+export interface ButtonProps extends Omit<ButtonContainerProps, 'ref'> {}
+
+export const Button = createComponent('button')({
+  displayName: 'Button',
+  Component: (
+    {
+      children,
+      icon,
+      colors,
+      iconPosition = icon ? (children ? 'start' : 'only') : undefined,
+      shouldMirrorIcon,
+      size = 'medium',
+      ...elemProps
+    }: ButtonProps,
+    ref,
+    Element
+  ) => {
+    return (
+      <BaseButton
+        as={Element}
+        ref={ref}
+        size={size}
+        icon={icon}
+        iconPosition={iconPosition}
+        type="button"
```

Comment:
```suggestion
```

Reply:
this is already added on base button https://github.com/Workday/canvas-kit/pull/2285/files#diff-417ee64d3f922776d0d7fc16914309849f3820955d4110431c675c0f4043d885R35